lib/ddtrace/context.rb

```diff
@@ -13,110 +13,84 @@ module Datadog
   # \Context, it will be related to the original trace.
   #
   # This data structure is thread-safe.
-  # rubocop:disable Metrics/ClassLength
   class Context
     # 100k spans is about a 100Mb footprint
     DEFAULT_MAX_LENGTH = 100_000
 
     attr_reader :max_length
 
-    # Initialize a new thread-safe \Context.
+    # Initialize a new \Context.
```

Comment:
Context stopped being thread safe. However, I've browsed the code and so far I haven't found a case where that was needed in the current implementation.

Would be good to have more eyes on this